version/version.go

```diff
@@ -23,29 +22,61 @@ var Version = "0.5.3"
 // such as "dev" (in development), "beta", "rc1", etc.
 var VersionPrerelease = "dev"
 
+// The metadata for the version, this is optional information to add around
+// a particular release.
+//
+// This has no impact on the ordering of plugins, and is ignored for non-human eyes.
+var VersionMetadata = ""
+
 // SDKVersion is used by the plugin set to allow Packer to recognize
 // what version of the sdk the plugin is.
-var SDKVersion = InitializePluginVersion(Version, VersionPrerelease)
+var SDKVersion = NewPluginVersion(Version, VersionPrerelease, VersionMetadata)
 
 // InitializePluginVersion initializes the SemVer and returns a version var.
 // If the provided "version" string is not valid, the call to version.Must
 // will panic. Therefore, this function should always be called in a package
 // init() function to make sure that plugins are following proper semantic
 // versioning and to make sure that plugins which aren't following proper
 // semantic versioning crash immediately rather than later.
+//
+// Deprecated: InitializePluginVersion does not support metadata out of the
+// box, and should be replaced by either NewRawVersion or NewPluginVersion.
 func InitializePluginVersion(vers, versionPrerelease string) *PluginVersion {
-	if vers == "" {
+	return NewPluginVersion(vers, versionPrerelease, "")
+}
+
+func NewRawVersion(rawSemVer string) *PluginVersion {
```

Comment:
Does it make sense to add comments to these two methods with examples of when to use them?

Reply:
Oh yeah good call, I forgot to document them, I'll add this thanks!